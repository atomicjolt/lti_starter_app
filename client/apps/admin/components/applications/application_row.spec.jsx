import React         from 'react';
import TestUtils     from 'react-dom/test-utils';
import { Provider }  from 'react-redux';
import Helper        from '../../../../specs_support/helper';
import ApplicationRow from './application_row';

describe('applications application row', () => {
  let result;
  let props;

  beforeEach(() => {
    props = {
      application: {
        id                          : 314159,
        name                        : 'SPECNAME',
        application_instances_count : 123
      },
      saveApplication: () => {}
    };

    result = TestUtils.renderIntoDocument(
      <Provider store={Helper.makeStore()}>
        <table><tbody>
          <ApplicationRow {...props} />
        </tbody></table>
      </Provider>
    );

  });

  it('button is clicked', () => {
    const button = TestUtils.findRenderedDOMComponentWithClass(result, 'i-settings');
    TestUtils.Simulate.click(button);
    expect(button).toBeDefined();
  });

  it('renders application instances count', () => {
    const span = TestUtils.findRenderedDOMComponentWithTag(result, 'span');
    expect(span.textContent).toContain('123');
  });

  it('renders application link', () => {
    const linkTag = TestUtils.findRenderedDOMComponentWithTag(result, 'a');
    expect(linkTag.innerHTML).toContain('SPECNAME');
  });

});
