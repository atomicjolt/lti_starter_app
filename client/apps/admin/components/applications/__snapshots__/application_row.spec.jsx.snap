// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`applications application row matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ApplicationRow
    application={
      Object {
        "application_instances_count": 1234,
        "id": 314159,
        "name": "SPECNAME",
      }
    }
    saveApplication={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <td>
          <Link
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/applications/314159/application_instances"
          >
            SPECNAME
          </Link>
        </td>,
        <td>
          <span>
            1234
          </span>
        </td>,
        <td>
          <button
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "transparent",
                "border": "none",
                "color": "grey",
                "cursor": "pointer",
                "fontSize": "1.5em",
              }
            }
          >
            <i
              className="i-settings"
            />
          </button>
          <Modal
            application={
              Object {
                "application_instances_count": 1234,
                "id": 314159,
                "name": "SPECNAME",
              }
            }
            closeModal={[Function]}
            isOpen={false}
            save={[Function]}
          />
        </td>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Link
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/applications/314159/application_instances"
          >
            SPECNAME
          </Link>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "SPECNAME",
            "onlyActiveOnIndex": false,
            "style": Object {},
            "to": "/applications/314159/application_instances",
          },
          "ref": null,
          "rendered": "SPECNAME",
          "type": [Function],
        },
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <span>
            1234
          </span>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 1234,
          },
          "ref": null,
          "rendered": 1234,
          "type": "span",
        },
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "border": "none",
                  "color": "grey",
                  "cursor": "pointer",
                  "fontSize": "1.5em",
                }
              }
            >
              <i
                className="i-settings"
              />
            </button>,
            <Modal
              application={
                Object {
                  "application_instances_count": 1234,
                  "id": 314159,
                  "name": "SPECNAME",
                }
              }
              closeModal={[Function]}
              isOpen={false}
              save={[Function]}
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <i
                className="i-settings"
              />,
              "onClick": [Function],
              "style": Object {
                "backgroundColor": "transparent",
                "border": "none",
                "color": "grey",
                "cursor": "pointer",
                "fontSize": "1.5em",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "i-settings",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "application": Object {
                "application_instances_count": 1234,
                "id": 314159,
                "name": "SPECNAME",
              },
              "closeModal": [Function],
              "isOpen": false,
              "save": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "td",
      },
    ],
    "type": "tr",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <td>
            <Link
              onlyActiveOnIndex={false}
              style={Object {}}
              to="/applications/314159/application_instances"
            >
              SPECNAME
            </Link>
          </td>,
          <td>
            <span>
              1234
            </span>
          </td>,
          <td>
            <button
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "border": "none",
                  "color": "grey",
                  "cursor": "pointer",
                  "fontSize": "1.5em",
                }
              }
            >
              <i
                className="i-settings"
              />
            </button>
            <Modal
              application={
                Object {
                  "application_instances_count": 1234,
                  "id": 314159,
                  "name": "SPECNAME",
                }
              }
              closeModal={[Function]}
              isOpen={false}
              save={[Function]}
            />
          </td>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              onlyActiveOnIndex={false}
              style={Object {}}
              to="/applications/314159/application_instances"
            >
              SPECNAME
            </Link>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "SPECNAME",
              "onlyActiveOnIndex": false,
              "style": Object {},
              "to": "/applications/314159/application_instances",
            },
            "ref": null,
            "rendered": "SPECNAME",
            "type": [Function],
          },
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span>
              1234
            </span>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 1234,
            },
            "ref": null,
            "rendered": 1234,
            "type": "span",
          },
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                onClick={[Function]}
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "border": "none",
                    "color": "grey",
                    "cursor": "pointer",
                    "fontSize": "1.5em",
                  }
                }
              >
                <i
                  className="i-settings"
                />
              </button>,
              <Modal
                application={
                  Object {
                    "application_instances_count": 1234,
                    "id": 314159,
                    "name": "SPECNAME",
                  }
                }
                closeModal={[Function]}
                isOpen={false}
                save={[Function]}
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <i
                  className="i-settings"
                />,
                "onClick": [Function],
                "style": Object {
                  "backgroundColor": "transparent",
                  "border": "none",
                  "color": "grey",
                  "cursor": "pointer",
                  "fontSize": "1.5em",
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "i-settings",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "application": Object {
                  "application_instances_count": 1234,
                  "id": 314159,
                  "name": "SPECNAME",
                },
                "closeModal": [Function],
                "isOpen": false,
                "save": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "td",
        },
      ],
      "type": "tr",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
